crates/transpiler/src/passes/commutation_cancellation.rs

```diff
@@ -252,24 +270,22 @@ pub fn cancel_commutations(
                     _ => unreachable!(),
                 };
 
-                let gate_angle = euler_one_qubit_decomposer::mod_2pi(total_angle, 0.);
+                let pi_multiple = total_angle / PI;
 
-                let new_op_phase: f64 = if gate_angle.abs() > _CUTOFF_PRECISION {
-                    dag.insert_1q_on_incoming_qubit((*new_op, &[total_angle]), cancel_set[0]);
-                    let Param::Float(new_phase) = new_op
-                        .definition(&[Param::Float(total_angle)])
-                        .unwrap()
-                        .global_phase()
-                        .clone()
-                    else {
-                        unreachable!();
-                    };
-                    new_phase
+                let mod4 = pi_multiple.rem_euclid(4.);
+                if mod4 < _CUTOFF_PRECISION || (4. - mod4) < _CUTOFF_PRECISION {
+                    // if the angle is close to a 4-pi multiple (from above or below), then it is
+                    // equal to the identity
```

Comment:
super-minor nitpick: it's not the angle that is equal to the identity, but the rotation gate :) or maybe the operator corresponding to this rotation gate